content/kb/troubleshooting-volume-readonly-or-io-error.md

```diff
@@ -0,0 +1,73 @@
+```
+/ # cd data
+/data # echo test > test
+sh: can't create test: I/O error
+```
+
+When running the `dmesg | grep EXT4-fs` in the related pod or the node host, the message like the following is shown:
+```
+......
+[1586907.286218] EXT4-fs (sdc): mounted filesystem with ordered data mode. Opts: (null)
+[1587396.152106] EXT4-fs warning (device sdc): ext4_end_bio:323: I/O error 10 writing to inode 12 (offset 0 size 4096 starting block 33026)
+[1587403.789877] EXT4-fs error (device sdc): ext4_find_entry:1455: inode #2: comm sh: reading directory lblock 0
+[1587404.353594] EXT4-fs warning (device sdc): htree_dirblock_to_tree:994: inode #2: lblock 0: comm ls: error -5 reading directory block
+[1587404.353598] EXT4-fs error (device sdc): ext4_journal_check_start:61: Detected aborted journal
+[1587404.355087] EXT4-fs (sdc): Remounting filesystem read-only
+......
+```
+
+When checking the event using `kubectl -n longhorn-system get event | grep <volume name>`, the event like the following is shown:
+```
+2m26s       Warning   DetachedUnexpectly       volume/pvc-342edde0-d3f4-47c6-abf6-bf8eeda3c32c               Engine of volume pvc-342edde0-d3f4-47c6-abf6-bf8eeda3c32c dead unexpectedly, reattach the volume
+```
+
+When checking for logs of the longhorn manager pods on the node the workload is running by running `kubectl -n longhorn-system logs <longhorn manager pod name> | grep <volume name>`, the following message is shown:
+
+```
+time="2021-01-05T11:20:46Z" level=debug msg="Instance handler updated instance pvc-342edde0-d3f4-47c6-abf6-bf8eeda3c32c-e-0fe2dac3 state, old state running, new state error"
+time="2021-01-05T11:20:46Z" level=warning msg="Instance pvc-342edde0-d3f4-47c6-abf6-bf8eeda3c32c-e-0fe2dac3 crashed on Instance Manager instance-manager-e-a1fd54e4 at shuo-cluster-0-worker-3, try to get log"
+......
+time="2021-01-05T11:20:46Z" level=warning msg="Engine of volume dead unexpectedly, reattach the volume" accessMode=rwo controller=longhorn-volume frontend=blockdev node=shuo-cluster-0-worker-3 owner=shuo-cluster-0-worker-3 state=attached volume=pvc-342edde0-d3f4-47c6-abf6-bf8eeda3c32c
+......
+time="2021-01-05T11:20:46Z" level=info msg="Event(v1.ObjectReference{Kind:\"Volume\", Namespace:\"longhorn-system\", Name:\"pvc-342edde0-d3f4-47c6-abf6-bf8eeda3c32c\", UID:\"69bb0f94-da48-4d15-b861-add435f25d00\", APIVersion:\"longhorn.io/v1beta1\", ResourceVersion:\"7466467\", FieldPath:\"\"}): type: 'Warning' reason: 'DetachedUnexpectly' Engine of volume pvc-342edde0-d3f4-47c6-abf6-bf8eeda3c32c dead unexpectedly, reattach the volume"
+```
+
+### Most common/Direct reason
+The mount point of the Longhorn volume becomes invalid once the Longhorn volume crashes unexpectedly. Then there is no way to read or write data in the Longhorn volume via the mount point.
+
+For Longhorn version earlier than v1.1.0, Longhorn will try to remount the volume automatically but it doesn't work well.
```

Comment:
`Longhorn will try to remount the volume automatically but the scenario it can handle is limited.`